packages/lesmis/src/api/queries/useDashboardData.js

```diff
@@ -6,8 +6,37 @@
 import { useQuery } from 'react-query';
 import { get } from '../api';
 
-export const useDashboardData = entityCode =>
-  useQuery(['dashboard', entityCode], () => get(`dashboard/${entityCode}`), {
+const getDrillDownCodes = dashboardItems =>
+  dashboardItems
+    .filter(d => !!d.drillDown)
+    .reduce((codes, d) => {
+      const entry = d.drillDown?.itemCodeByEntry;
+
+      if (entry) {
+        const values = Object.values(d.drillDown.itemCodeByEntry);
+        return [...codes, ...values];
+      }
+      return codes;
+    }, []);
+
+export const useDashboardData = entityCode => {
+  const query = useQuery(['dashboard', entityCode], () => get(`dashboard/${entityCode}`), {
     staleTime: 60 * 60 * 1000,
     refetchOnWindowFocus: false,
   });
+
```

Comment:
I've added the data processing in the api data layer to keep it out of the components. The processing is:
 - parse drilldown config
 - remove un-supported visual types